### PR TITLE
feat(#49): Multi-phase setup wizard + safe-operations instruction + scaffolding

### LINKED ISSUE #49: Multi-phase setup wizard overhaul + reusable scaffolding from Countdown-Clash patterns
## Overview

Overhaul the `/setup` wizard from a monolithic 2-stage process into a multi-phase, skip-gated experience that works for both brand-new and experienced users. Extract reusable patterns from the [Countdown-Clash](https://github.com/Grimblaz-and-Friends/Countdown-Clash) repo into workflow-template as shared instruction files and setup-generated scaffolding.

## Background: Countdown-Clash Asset Audit

Reviewed all `.github/` assets in Countdown-Clash to identify what's reusable vs project-specific:

### Already shared (via `chat.*Locations` settings)
- All 13 agents, 11 skills, 3 instruction files, hooks, prompts, cleanup scripts

### CC-specific (stays project-local)
- `copilot-instructions.md` / `architecture-rules.md` (game-domain content)
- Game-domain skills: `game-system-testing`, `game-visual-design`, `phaser-architecture`
- `validate-architecture.ps1` (CC's Systems/Scenes boundary checks)
- `standard-implementation-plan.md` (CC-customized plan with Phaser phases)
- Toolchain configs: eslint, prettier, tsconfig, vite, vitest, husky, package.json

### Reusable but not yet in workflow-template (candidates)
| Asset | What it is | Integration approach |
|-------|-----------|---------------------|
| `.vscode/settings.json` | Editor defaults (formatter, excludes) | Setup-generated scaffolding (Phase 5) |
| `.vscode/extensions.json` | Recommended extensions | Setup-generated scaffolding (Phase 5) |
| `.vscode/mcp.json` | Playwright MCP config | Conditional scaffolding (web projects only) |
| `.vscode/tasks.json` | Build/test tasks | Deferred — too project-specific currently |
| `.gitignore` copilot lines | `/.copilot-tracking/`, `/.copilot-tracking-archive/`, `screenshots/`, `/.playwright-mcp/` | Additive append to existing .gitignore |
| `browser-mcp.instructions.md` | Dev server check, screenshot rules, cleanup | Conditional scaffolding (web projects only) |
| File Operation Rules | "Never use Set-Content" guardrail | New shared instruction file |
| Issue Creation Rules | Priority labels, improvement-first decision | New shared instruction file |
| `Documents/` structure | `Design/`, `Decisions/`, `Development/` dirs | Setup-generated scaffolding |

## Design Decisions

| # | Decision | Choice | Rationale |
|---|----------|--------|-----------|
| 1 | Where reusable CC assets live | Setup-generated scaffolding | New repos get complete working environment from `/setup`; no manual copying |
| 2 | Setup flow model | Multi-phase with `ask_questions` skip gates | New users get guided; experienced users skip in seconds |
| 3 | File/Issue operation rules | Shared instruction: `safe-operations.instructions.md` | Universal guardrails for all repos using the workflow |
| 4 | Instruction file structure | Single combined file | Both rule sets are "safety/operations" concerns |
| 5 | Prerequisites | Auto-detect and report | Reduces friction; no manual checking |
| 6 | `.gitignore` strategy | Additive (append missing lines) | Respects existing project .gitignore |
| 7 | Scope | Focused: setup + instruction + scaffolding | `examples/` updates deferred to follow-up |

## New Setup Flow

```
/setup
├── Phase 0: Prerequisites Check (auto-detect)
│   ├── VS Code version ≥ 1.109.3
│   ├── pwsh on PATH
│   ├── git on PATH
│   └── gh CLI (optional, recommended)
│   → Report results, warn on missing, continue
│
├── Phase 1: User Setup (one-time, machine-level)
│   ├── Gate: "Is WORKFLOW_TEMPLATE_ROOT already set?" → Skip
│   ├── Detect OS
│   ├── Show WORKFLOW_TEMPLATE_ROOT set command
│   └── Show 5 VS Code settings to add
│
├── Phase 2: Project Basics
│   ├── Gate: "Does copilot-instructions.md already exist?" → Skip / Regenerate
│   ├── Project name
│   ├── What does it do?
│   ├── Primary language + version
│   ├── Framework + version
│   └── Database
│
├── Phase 3: Architecture & Conventions
│   ├── Gate: "Does architecture-rules.md already exist?" → Skip / Regenerate
│   ├── Architecture style
│   ├── Key conventions
│   └── Build tool
│
├── Phase 4: Commands
│   ├── Build command
│   ├── Run command
│   ├── Test command
│   ├── Lint/type-check command
│   └── Quick-validate command
│
├── Phase 5: Project Scaffolding
│   ├── Gate: "Do you want project scaffolding?" → Skip
│   ├── .gitignore — detect existing, append copilot-tracking lines
│   ├── .vscode/settings.json — formatter defaults, file/search excludes
│   ├── .vscode/extensions.json — recommended extensions
│   ├── If web project:
│   │   ├── .vscode/mcp.json — Playwright MCP config
│   │   ├── .github/instructions/browser-mcp.instructions.md
│   │   └── Dev server port question
│   └── Documents/ directory structure (Design/, Decisions/, Development/)
│
└── Summary: What was created/skipped
```

## System Touchpoints

| Component | Change |
|-----------|--------|
| `.github/prompts/setup.prompt.md` | Major rewrite — multi-phase structure with skip gates |
| `.github/instructions/safe-operations.instructions.md` | **New file** — file operation rules + issue creation patterns |
| `CUSTOMIZATION.md` | Update to reflect new phases and scaffolding options |
| `README.md` | Update setup description to match new phases |

**Customer surface**: CLI/prompt interaction via VS Code Copilot Chat (`/setup`)
**CE Gate approach**: Manual — run `/setup` in a test workspace and verify each phase works with both "skip" and "complete" paths

## Edge Cases

- `.gitignore` already has some but not all copilot-tracking lines → only append missing ones
- `.vscode/settings.json` already exists → ask: overwrite or skip
- User runs `/setup` mid-project (not from a fresh template) → all phases handle pre-existing files gracefully
- User has WORKFLOW_TEMPLATE_ROOT set but with a stale path → Phase 1 detects and offers to update
- User on macOS/Linux vs Windows → Phase 1 adapts env-var commands to OS

## Testing Scope

| Change Type | Unit | Integration | E2E |
|---|---|---|---|
| Setup prompt rewrite | N/A (markdown) | N/A | Manual walkthrough |
| safe-operations.instructions.md | N/A (markdown) | N/A | Verify agents load it |
| Scaffolding generation | N/A (prompt-driven) | N/A | Manual walkthrough |

No automated tests needed — this is configuration/documentation. Validation is manual: run `/setup` in a fresh workspace and verify output files are correct.

## CE Gate Readiness

1. **Customer surface**: VS Code Copilot Chat prompt (`/setup`)
2. **Tool availability**: Manual — run `/setup` in a test workspace
3. **Manual fallback**: Create files manually per CUSTOMIZATION.md
4. **Draft CE Gate scenarios**:
   - Run `/setup` as a brand-new user (no WORKFLOW_TEMPLATE_ROOT, no config files) — verify all phases execute and generate correct files
   - Run `/setup` as an experienced user (WORKFLOW_TEMPLATE_ROOT set, copilot-instructions.md exists) — verify skip gates work and only needed phases run
   - Run Phase 5 scaffolding on a repo with existing `.gitignore` — verify additive append behavior
   - Verify `safe-operations.instructions.md` is loaded by agents via `chat.instructionsFilesLocations`

## Acceptance Criteria

- [ ] New `safe-operations.instructions.md` exists with file-operation safety rules and issue-creation patterns extracted from CC
- [ ] `/setup` prompt rewritten with multi-phase structure (Phases 0–5)
- [ ] Each phase has an `ask_questions` skip gate (except Phase 0 which auto-runs)
- [ ] Phase 0 auto-detects prerequisites (VS Code, pwsh, git, gh)
- [ ] Phase 1 detects existing WORKFLOW_TEMPLATE_ROOT and offers to skip
- [ ] Phase 2 detects existing `copilot-instructions.md` and offers skip/regenerate
- [ ] Phase 3 detects existing `architecture-rules.md` and offers skip/regenerate
- [ ] Phase 5 generates `.vscode/settings.json`, `.vscode/extensions.json`, `.gitignore` additions
- [ ] Phase 5 conditionally generates `.vscode/mcp.json` and `browser-mcp.instructions.md` for web projects
- [ ] Phase 5 creates `Documents/` directory structure
- [ ] `.gitignore` generation is additive (appends only missing workflow-template lines)
- [ ] `CUSTOMIZATION.md` updated to reflect new phases
- [ ] `README.md` Quick Start updated to reflect new setup flow
- [ ] Existing `examples/` remain unchanged (deferred to follow-up)

### PR DESCRIPTION
**Pull request by Grimblaz**
## Summary

Overhauled the `/setup` wizard from a 2-stage flow (81 lines) to a 6-phase flow with skip gates (349 lines). Added `safe-operations.instructions.md` as a universal instruction file for file-operation safety and issue-creation patterns. Added Phase 5 project scaffolding generation.

## Motivation

- New users had no guidance on prerequisites (VS Code version, pwsh, git, gh CLI)
- Experienced users couldn't skip setup they'd already done — no skip gates existed
- Reusable patterns from Countdown-Clash (File Operation Rules, Issue Creation Rules) had no home in this repo
- No scaffolding step meant every new project had to manually configure `.vscode/` and `Documents/`

## Files Changed

| File | Change |
|------|--------|
| `.github/instructions/safe-operations.instructions.md` | **NEW** — Universal file-op safety rules + issue-creation patterns (priority labels, improvement-first decision rule, gh label create prerequisite) |
| `.github/copilot-instructions.md` | Instruction count 3→4 |
| `.github/prompts/setup.prompt.md` | Full rewrite: 81→349 lines, 2-stage → 6-phase with skip gates, working-dir guard, Phase 0 auto-detect, Phase 5 scaffolding |
| `CUSTOMIZATION.md` | Phase table in Quick Option, hook tip callout, new §7 scaffolding table |
| `README.md` | Quick Start updated to 6-phase, `safe-operations` added to config table, structure diagram updated |
| `Documents/Design/issue-49-setup-wizard-overhaul.md` | Design document |

## Phase Structure

| Phase | What it does | Skip available? |
|-------|-------------|-----------------|
| Phase 0 | Auto-detect prerequisites (`code --version`, `pwsh`, `git`, `gh`) | No |
| Phase 1 | One-time: `WORKFLOW_TEMPLATE_ROOT` + VS Code settings | Yes — if `WORKFLOW_TEMPLATE_ROOT` already set and valid |
| Phase 2 | Project basics (name, language, framework) | Yes — if `copilot-instructions.md` exists |
| Phase 3 | Architecture & conventions | Yes — if `architecture-rules.md` exists |
| Phase 4 | Build/run/test/lint/quick-validate commands | Yes — if Phases 2, 3, and 5 all skipped |
| Phase 5 | Scaffolding (.gitignore, .vscode/, Documents/) | Yes — opt-in |

## Validation Evidence

**Tier 1 — Quick-validate** (run twice: before and after review fixes):
```
Plan-Architect refs: 0 ✅
Janitor refs: 0 ✅
Agent count: 13 ✅
```

**Tier 7 — Code Review**: 3 independent Code-Critic passes, 19 findings across all passes, all adjudicated by Code-Review-Response:
- F3-1 (`chat.hookFilesLocations` validity): **REJECTED** — verified live in user settings and in Copilot Chat extension `enabledApiProposals`
- 17 findings **ACCEPTED** and fixed across 2 commits

**Key fixes applied from review**:
- `mode: ask` frontmatter (was `agent: ask` — invalid key)
- Phase 0 uses `code --version` (was Help > About menu action)
- Phase 4 conditional skip gate added
- Phase 5d `.vscode/mcp.json` overwrite guard added
- Phase 3 skip fallback in Generation section
- Improvement-first threshold aligned to `>1 day`
- Priority labels prerequisite + `gh label create` commands added
- `.NET IO static methods` added to FORBIDDEN list in safe-operations
- PowerShell equivalents replace POSIX `grep`/`wc` in copilot-instructions.md

**CE Gate**: ✅ CE Gate passed
- Scenario 1: `mode: ask` frontmatter verified ✅
- Scenario 2: All 6 phases (0–5) present at correct line numbers ✅
- Scenario 3: 5 skip gates wired at correct phases ✅
- Scenario 4: 4 instruction files present, all referenced example files exist ✅

_CE Gate surface: documentation/prompt wizard — no web UI or REST API. Structural validation performed via PowerShell file inspection._

**Process-Review (Track 2)**: No systemic gap found — review findings were implementation gaps in new content, not workflow process failures.

## Closes

Closes #49

## Summary by Sourcery

Overhaul the Copilot `/setup` prompt into a multi-phase, skippable setup wizard that also generates project scaffolding, and introduce shared safety/issue-creation instructions with corresponding documentation updates.

New Features:
- Introduce a 6-phase interactive setup wizard with prerequisite checks, skip gates, and project scaffolding generation for new repositories.
- Add a shared `safe-operations` instructions file defining cross-agent file operation safety rules and standardized issue-creation patterns, including priority labeling.
- Enable automatic generation of browser MCP configuration and documentation for web projects as part of the setup scaffolding flow.

Enhancements:
- Expand and reorganize project documentation (README, CUSTOMIZATION) to describe the new multi-phase wizard, scaffolding behavior, and available shared instructions.
- Update Copilot instructions to account for the new shared instruction file and replace POSIX-centric quick-validate commands with PowerShell equivalents for better cross-platform use.

Documentation:
- Add a design document detailing the setup wizard overhaul, phase structure, shared instruction file rationale, and review-driven fixes.

### CODE REVIEW

File: .github/instructions/safe-operations.instructions.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+````instructions
+# Safe Operations Instructions
+
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/instructions/safe-operations.instructions.md]: Fixed in eec43fb — removed the opening `` ````instructions `` fence and closing backtick pair. The file now starts with `# Safe Operations Instructions`, consistent with the other 3 instruction files.
